source/1_engine/rex_renderer_core/include/rex_renderer_core/commands/create_pipeline_state_cmd.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "rex_renderer_core/commands/render_cmd.h"
+#include "rex_renderer_core/resource_slot.h"
+
+namespace rex
+{
+  namespace renderer
+  {
+    namespace commands
+    {
+      struct CreatePipelineStateCommandDesc
+      {
+        CreatePipelineStateCommandDesc()
+            : input_layout(ResourceSlot::make_invalid())
+            , shader_program(ResourceSlot::make_invalid())
+            , rasterizer_state(ResourceSlot::make_invalid())
+            , blend_state(ResourceSlot::make_invalid())
+            , depth_stencil_state(ResourceSlot::make_invalid())
+            , num_render_targets(0)
+        {}
+
+        ResourceSlot input_layout;
+        ResourceSlot shader_program;
+        ResourceSlot rasterizer_state;
+        ResourceSlot blend_state;
+        ResourceSlot depth_stencil_state;
+        s32 num_render_targets;
+      };
+
+      class CreatePipelineState : public RenderCommand
+      {
+      public:
+        CreatePipelineState(CreatePipelineStateCommandDesc&& desc, ResourceSlot slot)
+            : RenderCommand()
+            , m_desc(rsl::move(desc))
+            , m_resource_slot(slot)
+        {
+        }
+
+        ~CreatePipelineState() override = default;
+
+        bool execute() override
+        {
+          return backend::create_pipeline_state_object(m_desc, m_resource_slot);
+        }
+
+      private:
+        CreatePipelineStateCommandDesc m_desc;
+        ResourceSlot m_resource_slot;
+      };
+    } // namespace commands
+  }   // namespace renderer
+} // namespace rex
+
+namespace rsl
+{
+  inline namespace v1
+  {
+    template <>
+    struct hash<rex::renderer::commands::CreatePipelineStateCommandDesc>
+    {
+      rsl::hash_result operator()(const rex::renderer::commands::CreatePipelineStateCommandDesc& createPipelineStateCommand) const
```

Comment:
This does not point to the content but to the slot id of each resource within the PSO, each slot id is unique so if the same value comes out that means it will point to the same object if all of the slot ids are the same the PSO will also be the same. Hashes are already stored within a hash table within the dx renderer class.

Reply:
linked to https://github.com/New-Creta/Rex/issues/263